src/UserSpaceInstrumentation/MachineCode.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef USER_SPACE_INSTRUMENTATION_MACHINE_CODE_TOOLING_H
+#define USER_SPACE_INSTRUMENTATION_MACHINE_CODE_TOOLING_H
+
+#include <cstdint>
+#include <vector>
+
+namespace orbit_user_space_instrumentation {
+
+// Close to trivial tooling to build up a vector of bytes used to write machine code in a somewhat
+// structured way.
+//
+// Usage example:
+// code.AppendBytes({0x48, 0xb8})
+//     .AppendImmediate64(address)
+//     .AppendBytes({0xff, 0xd0})
+//     .AppendBytes({0xcc});
+//
+// WriteTraceesMemory(pid, address_code, code.GetResultAsVector());
+//
+class MachineCode {
+ public:
+  MachineCode& AppendBytes(const std::vector<uint8_t>& data);
+  MachineCode& AppendImmediate64(uint64_t data);
+  MachineCode& AppendImmediate32(uint32_t data);
+  const std::vector<uint8_t>& GetResultAsVector() { return data_; }
```

Comment:
and the method can also now can be made const.

Reply:
done